doc/specs/Proto extensions-spec.md

```diff
@@ -0,0 +1,139 @@
+2. Generate profiles from the dynamic profile generators
+3. **Incorporate the additional provided json stubs** - these stubs could be:
+   1. modifications to existing profiles
+   2. full profiles
+   3. colour schemes/modifications to colour schemes
+4. Layer the user settings on top of all the profiles created in steps 1 through 3
+5. Validate the settings
+
+### Specifications of the json stubs
+
+As written above, the json stubs could be modifications to existing profiles, full profiles, or colour schemes.
+
+#### Modifications to existing profiles
+
+For modifications to existing profiles, the json stub would need to indicate which profile it wishes to modify.
+A straightforward way to do this would be by providing the name of the profile. Since this modification would
+occur at the step _before_ any user changes to names are layered, the names of the profiles are consistent and
+can be used for identification. For example, the name of a profile generated by the WSL profile generator is
+simply the argument to "wsl.exe -d \<name\>", the name of a profile generated by the Powershell profile generator
+would be "Powershell Core" or "Powershell (ARM)" etc depending on the version.
+
+We might run into the case where multiple json stubs modify the same profile and so they override each other. We
+will need some sort of hierarchy to determine which stub to ultimately use. My initial thought is that we could
+check the owner of the json file and see if it matches the profile they are trying to modify (so something like
+only Powershell can modify the Powershell profiles and only Ubuntu can modify the Ubuntu profile etc). Independent
+parties will need to create full profiles with the desired commandline argument instead.
+
+#### Full profile stubs
+
+Technically, full profile stubs do not need to contain anything (they could just be '\{\}'). However we should
+have some qualifying minimum criteria before we accept a stub as a full profile. I suggest that we only create
+new profile objects from stubs that contain at least the following
+
+* A name (and this name _should not_ match the names of existing profiles as described above, because that would cause us to interpret this profile as modifying an existing one)
+* A commandline argument
+* A unique GUID - this is so we avoid conflicts if several different creators name their profile the same thing
+
+As in the case of the dynamic profile generator, if we create a profile that did not exist before (i.e. does not
+exist in the user settings), we need to add the profile to the user settings file and re-save that file. 
+
+#### Colour schemes
+
+Once again, we could run into the problem of several json stubs wanting to modify the same colour scheme. We
+will require a hierarchy here as well. Though since the 'owner' of a colour scheme is a lot more dubious, I wonder
+if we should only allow creation of new colour schemes for now and ignore requests to modify existing colour
+schemes. 

```

Comment:
And if someone wants to add a new colour scheme, we should probably append their unique identifier to the name (so we don't allow someone to lay claim to the name "Best Colour Scheme Ever" or something)

Reply:
I don't like the idea of appending a unique identifier to the name, unless we can rename it. If that's the case, please go into more details here. Maybe we need to add a "source" to the imported color scheme?

Reply:
That would mean we would need to revamp how we interpret colour schemes - for example if someone adds a colour scheme named "Campbell", when we parse out what that "Campbell" means we will also need to match the source before we can apply the scheme (and that means we will also need to define a behaviour for when the source is unspecified). 

I feel like appending a unique identifier to the name gives us the same functionality without needing to mess with everywhere else a colour scheme is interpreted.

Reply:
For now, we will just let multiple colour schemes with the same name override each other.